contracts/contracts/strategies/plume/RoosterAMOStrategy.sol

```diff
@@ -0,0 +1,1301 @@
+    /// @param _maverickPosition Address of the Maverick's position contract
+    /// @param _maverickQuoter Address of the Maverick's Quoter contract
+    /// @param _mPool Address of the Aerodrome concentrated liquidity pool
+    /// @param _upperTickAtParity Bool when true upperTick is the one where the
+    ///        price of OETHp and WETH are at parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _oethpAddress,
+        address _liquidityManager,
+        address _poolLens,
+        address _maverickPosition,
+        address _maverickQuoter,
+        address _mPool,
+        bool _upperTickAtParity,
+        address _votingDistributor,
+        address _poolDistributor
+    ) initializer InitializableAbstractStrategy(_stratConfig) {
+        require(
+            address(IMaverickV2Pool(_mPool).tokenA()) == _wethAddress,
+            "Only WETH supported as tokenA"
+        );
+        require(
+            address(IMaverickV2Pool(_mPool).tokenB()) == _oethpAddress,
+            "Only OETHp supported as tokenB"
+        );
+        require(
+            _liquidityManager != address(0),
+            "LiquidityManager zero address not allowed"
+        );
+        require(
+            _maverickQuoter != address(0),
+            "Quoter zero address not allowed"
+        );
+        require(_poolLens != address(0), "PoolLens zero address not allowed");
+        require(
+            _maverickPosition != address(0),
+            "Position zero address not allowed"
+        );
+        require(
+            _votingDistributor != address(0),
+            "Voting distributor zero address not allowed"
+        );
+        require(
+            _poolDistributor != address(0),
+            "Pool distributor zero address not allowed"
+        );
+
+        uint256 _tickSpacing = IMaverickV2Pool(_mPool).tickSpacing();
+        require(_tickSpacing == 1, "Unsupported tickSpacing");
+        // -1 is the tick where the contract deposits liquidity to
+        tickNumber = -1;
+
+        // tickSpacing == 1
+        (sqrtPriceTickLower, sqrtPriceTickHigher) = TickMath.tickSqrtPrices(
+            _tickSpacing,
+            tickNumber
+        );
+        sqrtPriceAtParity = _upperTickAtParity
+            ? sqrtPriceTickHigher
+            : sqrtPriceTickLower;
+
+        WETH = _wethAddress;
+        OETHp = _oethpAddress;
+        liquidityManager = IMaverickV2LiquidityManager(_liquidityManager);
+        poolLens = IMaverickV2PoolLens(_poolLens);
+        maverickPosition = IMaverickV2Position(_maverickPosition);
+        quoter = IMaverickV2Quoter(_maverickQuoter);
+        mPool = IMaverickV2Pool(_mPool);
+        votingDistributor = IVotingDistributor(_votingDistributor);
+        poolDistributor = IPoolDistributor(_poolDistributor);
+
+        require(address(mPool.tokenA()) == WETH, "WETH not TokanA");
+        require(address(mPool.tokenB()) == OETHp, "OETHp not TokanB");
```

Comment:
This seems a duplicate with the previous requires, at the beginning of the constructor.
> require(address(IMaverickV2Pool(_mPool).tokenA()) == _wethAddress, "Only WETH supported as tokenA");
> require(address(IMaverickV2Pool(_mPool).tokenB()) == _oethpAddress,"Only OETHp supported as tokenB");

Reply:
nice catch: https://github.com/OriginProtocol/origin-dollar/pull/2499/commits/14f8ba4cb4cc12e60dec3f5422c3ceaa80faeb03